components/UI/FilterMenu/FilterReasoningMenu.tsx

```diff
@@ -44,6 +44,19 @@ export interface FilterReasoningMenuProps {
   onChange: (_option: ReasoningFilterMenuOption) => void;
 }
 
+export const getReasoningFilter = (
+  reasoningFilterMenuOption: ReasoningFilterMenuOption
+) => {
+  reasoningFilterMenuOption.type;
+  if (reasoningFilterMenuOption.type === "channel") {
+    return undefined;
+  }
+
+  if (reasoningFilterMenuOption.type === "reason") {
+    return reasoningFilterMenuOption.value;
+  }
+};
```

Comment:
kapatmdan once return; yazman yeterli. en ustteki kosula girmezse bruaya duser.

Reply:
Bu kod parcasini baska bir dosyadan tasimistim. Okumadan tasimisim. Okuyunca switch/case'in daha uygun oldugunu gordum. Implemente edip kodu pushladim.